src/main/java/soot/toolkits/scalar/ConstantValueToInitializerTransformer.java

```diff
@@ -173,6 +180,22 @@ public void transformClass(SootClass sc) {
     }
   }
 
+  private boolean isInstanceFieldAssignedConstantInBody(SootField sf, Constant constant, Body body) {
+    for (Unit u : body.getUnits()) {
```

Comment:
wouldn't it be better to cache the results for the field and constant rather than go through every time this loop?

Reply:
If `SootClass.getFields()` and `SootClass.getMethods()` return distinct member fields and methods respectively, then this check is performed once per distinct pairs of instance field (with a constant initializer) and (constructor) method. 

Caching could then benefit only for multiple copies of the same `ConstantValueTag` in `getTags()`. Could that be the case?

Reply:
Further, a field may be initialized to (at most) one constant. Note that a constant initialization in a super-class that is overridden in a sub-class will be handled separately- once for constructor(s) of super-class, again for constructor(s) of sub-class. So, the only benefit of caching here could be due to duplicate tags for the same constant value?

If so, handling/fixing that may be considered outside the scope of this PR?

Reply:
Sounds reasonable to me.